code/modules/projectiles/guns/misc/syringe_gun.dm

```diff
@@ -62,11 +65,17 @@
 			to_chat(user, "<span class='notice'>You load [A] into \the [src].</span>")
 			syringes += A
 			recharge_newshot()
+			update_icon_state()
+			playsound(src, load_sound, 40)
 			return TRUE
 		else
 			to_chat(user, "<span class='warning'>[src] cannot hold more syringes!</span>")
 	return FALSE
 
+/obj/item/gun/syringe/update_icon_state()
+	if (max_syringes == 1)
+		icon_state = "syringegun[syringes.len > 0 ? "" : "_empty"]"
```

Comment:
there's a bunch of subtypes below that'll have the wrong sprite after being loaded now.

also, should probably just make this an overlay.

Reply:
Forgot about the stuff other than the rapid syringe gun. What would you recommend here? A flag that all subtypes tick off, or explicitly checking that we're not in a subtype?